### PR TITLE
[blockly] finally the right implementation for parsing date formats

### PR DESCRIPTION
**Pull request by stefan-hoehn**
The original implementation oh_zdt_fromText was awkward as it was based on length detection. Now it uses regex expressions that check the according format

A special thanks goes to Hristo who implemented all the regex's for me <3 

Also-by: @htrendev
Signed-off-by: Stefan Höhn <stefan@andreaundstefanhoehn.de>

### CODE REVIEW

**Comment by stefan-hoehn** [comment]
@ghys I am sure you appreciate this version much more than the original version ;-) Sorry that it took so long :-)

File: bundles/org.openhab.ui/web/src/assets/definitions/blockly/utils.js [hunk 1]
```diff
@@ -44,19 +44,27 @@ export function addDateSupport () {
...
-      `    case 28: return ${zdt}.parse(datetime.slice(0,26) + ':' + datetime.slice(26,28), dtf.ofPattern('yyyy-MM-dd HH:mm:ss.SSSSz'));`,
+      '  var regex_date =             /^\\d{4}-\\d{2}-\\d{2}$/; ',
+      '  var regex_date_time_min =    /^\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}$/; ',
+      '  var regex_date_time_sec =    /^\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}$/; ',
+      '  var regex_date_time_sec_tz = /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}[+-]\\d{2}:\\d{2}$/; ',
+      '  var regex_date_time_ms =     /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}\\.\\d{3}$/;   ',
+      '  var regex_date_time_us =     /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}\\.\\d{6}$/;  ',
+      '  var regex_date_time_ms_tz =  /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}\\.\\d{3}[+-]\\d{2}:\\d{2}$/; ',
+      '  var regex_date_time_us_tz =  /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}\\.\\d{6}[+-]\\d{2}:\\d{2}$/; ',
+      '  var regex_oh =               /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}\\.\\d{3}[+-]\\d{4}$/;   ',
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/assets/definitions/blockly/utils.js:57]: I don't suppose these dangling EOL spaces were intentional?

**Review by ghys — COMMENTED** [review]
Yes it's far more elegant and robust in my opinion ;) I'll trust you have successfully tested the regexes? :)

One obligatory cosmetic remark below:

File: bundles/org.openhab.ui/web/src/assets/definitions/blockly/utils.js (hunk 1 above)
Comment (stefan-hoehn) [reply bundles/org.openhab.ui/web/src/assets/definitions/blockly/utils.js]: Oh yes, definitely. I do not trust any regex right away ;-) 

No, the spaces weren't intended. I removed them.

**Review by ghys — APPROVED** [review]

**Pull request merged**